iceberg/src/main/scala/org/apache/spark/sql/delta/icebergShaded/IcebergConverter.scala

```diff
@@ -155,21 +159,79 @@ class IcebergConverter(spark: SparkSession)
     }
   }
 
+  /**
+   * Convert the specified snapshot into Iceberg for the given catalogTable
+   * @param snapshotToConvert the snapshot that needs to be converted to Iceberg
+   * @param catalogTable      the catalogTable this conversion targets.
+   * @return Converted Delta version and commit timestamp
+   */
+  override def convertSnapshot(
+      snapshotToConvert: Snapshot, catalogTable: CatalogTable): Option[(Long, Long)] = {
+    if (!UniversalFormat.icebergEnabled(snapshotToConvert.metadata)) {
+      return None
+    }
+    convertSnapshot(snapshotToConvert, None, catalogTable)
+  }
+
+  /**
+   * Convert the specified snapshot into Iceberg when performing an OptimisticTransaction
+   * on a delta table.
+   * @param snapshotToConvert the snapshot that needs to be converted to Iceberg
+   * @param txnOpt            the transaction that triggers the conversion. It must
+   *                          contain the catalogTable this conversion targets.
+   * @return Converted Delta version and commit timestamp
+   */
+  override def convertSnapshot(
+      snapshotToConvert: Snapshot,
+      txnOpt: Option[OptimisticTransactionImpl]): Option[(Long, Long)] = {
+    if (!UniversalFormat.icebergEnabled(snapshotToConvert.metadata)) {
+      return None
+    }
+    def logNoTableWarning(snapshot: Snapshot, msg: String) = {
+      logWarning(msg)
+      recordDeltaEvent(
+        snapshotToConvert.deltaLog,
+        "delta.iceberg.conversion.skipped.emptyCatalogTable",
+        data = Map(
+          "version" -> snapshotToConvert.version
+        )
+      )
+    }
+    txnOpt match {
+      case Some(txn) =>
+        txn.catalogTable match {
+          case Some(table) => convertSnapshot(snapshotToConvert, txnOpt, table)
+          case _ => logNoTableWarning(
+            snapshotToConvert,
+            s"CatalogTable for table ${snapshotToConvert.deltaLog.tableId} " +
+              s"is empty in txn. Skip iceberg conversion.")
+          None
```

Comment:
weird indent there...
```suggestion
          case _ => 
            logNoTableWarning(
              snapshotToConvert,
              s"CatalogTable for table ${snapshotToConvert.deltaLog.tableId} " +
                s"is empty in txn. Skip iceberg conversion.")
            None
```
(again below)